examples/demo-rollup/src/test_rpc.rs

```diff
@@ -98,18 +96,34 @@ fn test_helper(data: String, expected: &str, port: u16) {
 
         let ledger_rpc_module = ledger_rpc::get_ledger_rpc::<i32, i32>(ledger_db.clone());
 
+        let actual_port_placeholder: Arc<AtomicU16> = Arc::new(AtomicU16::new(0));
+        let actual_port_placeholder_clone = actual_port_placeholder.clone();
         rt.spawn(async move {
             let server = jsonrpsee::server::ServerBuilder::default()
                 .build([address].as_ref())
                 .await
                 .unwrap();
+            let actual_address = server.local_addr().unwrap();
+            actual_port_placeholder_clone.store(actual_address.port(), Ordering::Relaxed);
             let _server_handle = server.start(ledger_rpc_module).unwrap();
             tx_start.send("server started").unwrap();
             rx_end.await.unwrap();
         });
 
         rx_start.await.unwrap();
 
+        for _ in 0..1000 {
+            if actual_port_placeholder.load(Ordering::Relaxed) != 0 {
+                break;
+            }
+            std::thread::sleep(std::time::Duration::from_millis(10));
```

Comment:
It seems like it would be simpler and faster to use a oneshot channel here. Is there a reason we need to use an atomic with manual synchronization instead?

Reply:
I haven't noticed performance improvement, but it is simpler. Fix pushed